zcash_client_backend/src/data_api/chain.rs

```diff
@@ -277,6 +279,7 @@ where
     ParamsT: consensus::Parameters + Send + 'static,
     BlockSourceT: BlockSource,
     DbT: WalletWrite,
+    <DbT as WalletRead>::AccountId: Clone + Default + Eq + Send + 'static,
```

Comment:
What requires the `Default` bound here? I think that `Default` is an absolutely terrible trait whose only purpose is to cause subtle and hard-to-diagnose bugs, so unless it's a dependency of some upstream library and there's no workaround, I ban it from our codebase.

Reply:
I guess you found where it's necessary and commented.